layouts/partials/summary.html

```diff
@@ -1,12 +1,22 @@
 <article class="list-article">
     {{ if (or .Params.featured_image .Params.featured_image_url) }}
-        {{ $img_src := "" }}
-        {{ if .Params.featured_image }}
-            {{ $img_src = (partial "functions/imgURL" (dict "page" .Page "name" (.Params.featured_image) "command" "Resize" "options" "230x")) }}
-        {{ else }}
-            {{ $img_src = .Params.featured_image_url }}
-        {{ end }}
-        <div class="col25 article-featured-image"><a href="{{ .Permalink }}"><img class="image" src="{{ $img_src }}" alt="{{ .Title }}"></a></div>
+        <div class="col25 article-featured-image">
+            <a href="{{ .Permalink }}">
+                {{ if .Params.featured_image }}
+                    <img srcset="
+                            {{- partial "functions/imgURL" (dict "page" .Page "name" .Params.featured_image "command" "Resize" "options" "280x") }} 280w,
+                            {{- partial "functions/imgURL" (dict "page" .Page "name" .Params.featured_image "command" "Resize" "options" "375x") }} 375w,
+                            {{- partial "functions/imgURL" (dict "page" .Page "name" .Params.featured_image "command" "Resize" "options" "684x") }} 684w,
+                            {{- partial "functions/imgURL" (dict "page" .Page "name" .Params.featured_image "command" "Resize" "options" "874x") }} 874w,
+                            {{- partial "functions/imgURL" (dict "page" .Page "name" .Params.featured_image "command" "Resize" "options" "230x") }} 230w
+                        "
+                        sizes="(max-width: 320px) 280px, (max-width: 425px) 375px, (max-width: 768px) 684px, (max-width: 979px) 874px, 230px"
+                        src="{{ (partial "functions/imgURL" (dict "page" .Page "name" .Params.featured_image)) }}">
+                {{ else }}
+                    <img class="image" src="{{ .Params.featured_image_url }}" alt="{{ .Title }}">
+                {{ end }}
+            </a>
```

Comment:
Not quite sure what you mean by your suggestion to be honest.

I agree that this doesn't feel very DRY, though. However, I am not sure if we can necessarily do better. This may just be characteristic of Go templates…  
Most of the arguments to the partial call are actually specific to this particular case. I guess we could implement some abstraction that generates those lines but the problem then is that the sizes are specific to the case. And I don't even want to image what the Go template syntax for the necessary function would have to look like. :D

Reply:
Thus, even though it feels repeated, this is the clearest and most readable and maintainable approach imo. At least for the moment, this may change with #373, I have no clue what that will look like then.

Reply:
Ok, at second glance, I can see that they are clearly different and hard to abstract. Then this is fine of course!